packages/@ember/-internals/glimmer/lib/resolver.ts

```diff
@@ -95,7 +95,7 @@ export default class RuntimeResolver implements IRuntimeResolver<OwnedTemplateMe
   } = BUILTIN_MODIFIERS;
 
   // supports directly imported late bound layouts on component.prototype.layout
-  private templateCache: Map<Owner, Map<TemplateFactory, OwnedTemplate>> = new Map();
```

Comment:
`WeakMap` is more appropriate for `templateCache` I think, as owner destroyed all cache will be destroyed along with it

Reply:
Actually no, we absolutely want a `Map` here. This actually _resolved_ a memory leak in the 3.1 series.

We _never_ want to release templates during the lifetime of the application (which is what this resolver is scoped to).